docs/ref/models/conditional-expressions.txt

```diff
@@ -221,3 +221,17 @@ What if we want to find out how many clients there are for each
     ...     )
     ... )
     {'regular': 2, 'gold': 1, 'platinum': 3}
+
+Conditional filter
+------------------
+
+When a conditional expression returns a boolean value, it is possible to use
+the expression directly in a filter expression. This means it will not be
+returned as a column in the query, but will still be used to filter results::
+
+    >>> Client.objects.filter(Case(
```

Comment:
I'm not really sold on this example, because it's contrived (and you'd use an IN in your filter instead, I think).